src/webgpu/util/conversion.ts

```diff
@@ -2234,6 +2234,18 @@ export function isFloatType(ty: Type): boolean {
   return false;
 }
 
+/**
+ * @returns if `ty ` accepts a floating point value. abstract-int does not because it uses BigInt
```

Comment:
Is this actually querying whether `ScalarType.create()` accept a non-integer `number` ?

Given the description and surrounding code, I would have expected this to be whether a WGSL a floating-point type (`f32`, `f16`, `abstract-float`) type can be converted to `ty`.

For your usage, would `Number.isInteger(value) || isFloatType(scalarTypeOf(ty))` work?

Reply:
almost.  `isFloatType` only works for scalar types not composite types. add `isFloatTypeOrCompositeFloatType` ... or should I not be testing composite types here?